src/test/java/com/example/onlinebookstore/service/impl/BookServiceImplTest.java

```diff
@@ -0,0 +1,185 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.anyLong;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import com.example.onlinebookstore.dto.book.BookResponseDto;
+import com.example.onlinebookstore.dto.book.CreateBookRequestDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.BookMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.BookRepository;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTest {
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+    @Mock
+    private BookRepository bookRepository;
+    @Mock
+    private BookMapper bookMapper;
+
+    @Test
+    @DisplayName("""
+            save() works properly and returns dto
+            """)
+    void save_WithValidRequestBookDto_ShouldReturnCorrectDto() {
+        Book book = getDefaultBook();
+        BookResponseDto expected = getDefaultBookResponseDto();
+        CreateBookRequestDto createBookRequestDto = getDefaultCreateBookRequestDto();
+        when(bookMapper.toEntity(createBookRequestDto)).thenReturn(book);
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.toDto(book)).thenReturn(expected);
+
+        BookResponseDto actual = bookService.save(createBookRequestDto);
+
+        EqualsBuilder.reflectionEquals(actual, expected, "id");
+    }
+
+    @Test
+    @DisplayName("""
+            Verify the book title with valid ID was returned
+            """)
+    void findById_WithValidBookId_ShouldReturnValidBook() {
+        Book book = getDefaultBook();
+        String expected = book.getTitle();
+        BookResponseDto bookResponseDto = getDefaultBookResponseDto();
+        when(bookRepository.findById(anyLong())).thenReturn(Optional.of(book));
+        when(bookMapper.toDto(book)).thenReturn(bookResponseDto);
+
+        String actual = bookService.findById(book.getId()).getTitle();
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+            findById() should return Exception with invalid ID
```

Comment:
```suggestion
            Find bood by id, should return Exception with invalid ID
```
make more human-readable names